docs/source/debuggin.rst

```diff
@@ -0,0 +1,136 @@
+ 
+-``RuntimeWarning: Primal solution violates constraint: 1.0000149786 is greater than 1`` - this warning is typically seen in dual infeasible models, see *Dual Infeasible Models* below for more tips on debugging a dual infeasible model.
+
+-``RuntimeWarning: Dual cost nan does not match primal cost 1.00122315152`` - this warning is typically seen in dual infeasible models, see *Dual Infeasible Models* below for more tips on debugging a dual infeasible model.
+
+-``RuntimeWarning: final status of solver 'cvxopt' was 'unknown', not 'optimal’`` or - this is the most difficult warning to debug. It can be thrown when attempting to solve a dual infeasible model or a primal infeasible model. See *Dual Infeasible Models* and *Primal Infeasible Models* below for more information.
+
+-``RuntimeWarning: final status of solver 'mosek' was 'DUAL_INFEAS_CER', not 'optimal’`` - this error is thrown when attempting to solve a dual infeasible model with MOSEK,  see *Dual Infeasible Models* below for more tips on debugging a dual infeasible model.
+
+
+Dual Infeasible Models
+=============
+
+A dual infeasible error typically means that more than one unique solution meets the objective and satisfies the constraints. Usually, this means that one or more variables are not sufficiently constrained.  When solving with ``mosek``, the error message will tell you that the solution is dual-infeasible.  When solving with ``cvxopt``, the error for a dual-feasible solution will usually display as a ``Rank`` error or ``cvxopt`` will return ``unknown``.  An example of a dual-feasible model is shown below. This model is dual-infeasible because there are multiple values of ``x`` and ``y`` that satisfy the constraint set even though the obvious value of the objective should be 1.
+ 
+ .. code-block:: python
+ 
+     from gpkit import Variable, Model
+     x = Variable("x")
+     y = Variable("y")
+     m = Model(x*y, [x*y >= 1])
+     m.solve()
+ 
+Note: When solving with ``mosek`` this model will actually solve because ``mosek`` is robust enough to handle some dual-feasible problems. The following example is a model that ``mosek`` labels as dual-feasible and will not solve. 
+ 
+ .. code-block:: python
+ 
+     from gpkit import Variable, Model
+     x = Variable("x")
+     y = Variable("y")
+     m = Model(x**0.01 * y, [x*y >= 1])
+     m.solve()
+ 
+While this model is very similar to the previous model, ``mosek`` is unable to solve this model and labels it as dual-feasible.
+
+Another common cause of dual-infeasability is a model’s objective applying pressure on a variable in an unexpected direction. If the variable is not bounded in this direction, its value will be pushed to either zero or infinity. This usually results in the solver returning a final status of dual-infeasible. A simple example is given below. ``x`` has no lower bound, and the objective is to minimize ``x``, so the solver pushes ``x`` towards zero and returns dual infeasible.
```

Comment:
"unexpected" in first sentence is not precise.

Just come out and say it's caused by variables being pushed to zero or infinity.